org.eclipse.zest.layouts/src/org/eclipse/zest/layouts/algorithms/SpringLayoutAlgorithm.java

```diff
@@ -790,21 +792,24 @@ private int numRelations(Object src, Object dest) {
 	 */
 	private double avgWeight(Object src, Object dest) {
 		String key = src.toString() + dest.toString();
-		Double avgWeight = (Double) srcDestToRelsAvgWeightMap.get(key);
-		double doubleWeight = (avgWeight == null) ? 1 : avgWeight.doubleValue();
-		return doubleWeight;
+		Double avgWeight = srcDestToRelsAvgWeightMap.get(key);
+		return (avgWeight == null) ? 1 : avgWeight.doubleValue();
 	}
 
 	@Override
 	protected boolean isValidConfiguration(boolean asynchronous, boolean continueous) {
-		if (asynchronous && continueous)
+		if (asynchronous && continueous) {
 			return true;
-		else if (asynchronous && !continueous)
+		}
+		if (asynchronous && !continueous) {
 			return true;
-		else if (!asynchronous && continueous)
+		}
+		if (!asynchronous && continueous) {
 			return false;
-		else if (!asynchronous && !continueous)
+		}
+		if (!asynchronous && !continueous) {
 			return true;
+		}
 
 		return false;
```

Comment:
That can probably also be simplified to `return asynchronous || !continueous`

Reply:
I found this code extra spooky and was to afraid to change it. But you are totally right. Changed it accordingly.